sdk/spring/spring-cloud-azure-resourcemanager/src/main/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ResourceCrud.java

```diff
@@ -29,18 +30,20 @@ public interface ResourceCrud<T, K> {
     boolean exists(K key);
 
     /**
-     * Creates a value for a given key.
+     * Creates a value for a given key and properties.
      *
      * @param key The key.
+     * @param properties The properties for the Azure resource creation.
      * @return The created value.
      */
-    T create(K key);
+    T create(K key, P properties);
 
     /**
-     * Gets or creates a value for a given key.
+     * Gets or creates a value for a given key and properties.
      *
      * @param key The key.
+     * @param properties The properties for the Azure resource creation.
      * @return The retrieved or created value.
      */
-    T getOrCreate(K key);
+    T getOrCreate(K key, P properties);
```

Comment:
Should we add two other methods with properties provided, instead of changing the origin two?

Reply:
And for these, we provide default implementation, without the need to change in every implementation?